python/pyiceberg/catalog/glue.py

```diff
@@ -0,0 +1,202 @@
+        else:
+            raise NoSuchTableError(f"Table property {METADATA_LOCATION} is missing")
+
+        file = io.new_input(metadata_location)
+        metadata = FromInputFile.table_metadata(file)
+        return Table(
+            identifier=(glue_table['DatabaseName'], glue_table['Name']),
+            metadata=metadata,
+            metadata_location=metadata_location
+        )
+
+    def _iceberg_to_glue(self, iceberg_table):
+        # TODO
+        pass
+
+    def _construct_parameters(self, metadata_location: str) -> Dict[str, str]:
+        properties = {"table_type": "ICEBERG", "metadata_location": metadata_location}
+        return properties
+
+    def _default_warehouse_location(self, database_name: str, table_name: str):
+        try:
+            response = self.client.get_database(Name=database_name)
+        # TODO: handle response and errors
+        except:
+            raise NoSuchNamespaceError("Database not found")
+
+        if "LocationUri" in response["Database"]:
+            return f"{response['Database']['LocationUri']}/table_name"
+
+        # TODO: should extract warehouse path from the properties and handle potential errors
+        return f"{self.properties['warehouse_path']}/{database_name}.db/{table_name}"
+
+    def _resolve_table_location(self, location: Optional[str], database_name: str, table_name: str):
+        if not location:
+            return self._default_warehouse_location(database_name, table_name)
+        return location
+
+    def _write_metadata(self, metadata: TableMetadata, io: FileIO, metadate_path: str):
+        ToOutputFile.table_metadata(metadata, io.new_output(metadate_path))
+
+    # tested on pre-existing database
+    def create_table(
+            self,
+            identifier: Union[str, Identifier],
+            schema: Schema,
+            location: Optional[str] = None,
+            partition_spec: PartitionSpec = UNPARTITIONED_PARTITION_SPEC,
+            sort_order: SortOrder = UNSORTED_SORT_ORDER,
+            properties: Properties = EMPTY_DICT,
+    ) -> Table:
+        database_name, table_name = self.identifier_to_tuple(identifier)
+
+        location = self._resolve_table_location(location, database_name, table_name)
+        # TODO: give it correct path based on java version of glueCatalog
+        metadata_location = f"{location}/metadata/{uuid.uuid4()}.metadata.json"
+        metadata = new_table_metadata(
+            location=location, schema=schema, partition_spec=partition_spec, sort_order=sort_order,
+            properties=properties
+        )
+        io = load_file_io({**self.properties, **properties}, location=location)
+        self._write_metadata(metadata, io, metadata_location)
+        try:
+            create_table_response = self.client.create_table(
+                DatabaseName=database_name,
+                TableInput={
+                    'Name': table_name,
+                    'Description': '',  # To be fixed
```

Comment:
can take from description in `metadata`

Reply:
Thank you for your suggestions. Just to confirm, here the `description` should be one of the props in `properties`. So I think maybe  I can directly extract description from the `properties`?